File: SDKs/amazonaws.com/codestar/2017-04-19/python/src/sdk/models/shared/toolchain.py
from dataclasses import dataclass, field
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from . import *


@dataclass_json
@dataclass
class Toolchain:
    r"""Toolchain
    The toolchain template file provided with the project request. AWS CodeStar uses the template to provision the toolchain stack in AWS CloudFormation.
    """
    
    source: ToolchainSource = field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('source') }})
    role_arn: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('roleArn') }})
    stack_parameters: Optional[dict[str, str]] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('stackParameters') }})
    

File: SDKs/amazonaws.com/codestar/2017-04-19/typescript/src/sdk/models/shared/toolchain.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { ToolchainSource } from "./toolchainsource";



// Toolchain
/** 
 * The toolchain template file provided with the project request. AWS CodeStar uses the template to provision the toolchain stack in AWS CloudFormation.
**/
export class Toolchain extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=roleArn" })
  roleArn?: string;

  @SpeakeasyMetadata({ data: "json, name=source" })
  source: ToolchainSource;

  @SpeakeasyMetadata({ data: "json, name=stackParameters" })
  stackParameters?: Map<string, string>;
}
